core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -951,15 +951,22 @@ class KafkaApis(val requestChannel: RequestChannel,
               new OffsetFetchResponse(Errors.NONE, authorizedPartitionData.asJava, header.apiVersion)
             }
           } else {
-            val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
-              .partition(authorizeTopicDescribe)
-            val (error, authorizedPartitionData) = coordinator.handleFetchOffsets(offsetFetchRequest.groupId,
-              Some(authorizedPartitions))
+            val (authorizedPartitions, unauthorizedPartitions) =
+              offsetFetchRequest.partitions.asScala.partition(authorizeTopicDescribe)
+            val (error, authorizedPartitionData) =
+              coordinator.handleFetchOffsets(offsetFetchRequest.groupId, Some(authorizedPartitions))
             if (error != Errors.NONE)
               offsetFetchRequest.getErrorResponse(error)
             else {
+              val authorizedAndValidatedPartitionData = authorizedPartitionData.map {
+                case (topicPartition, offsetInfo) =>
+                  if (metadataCache.getPartitionInfo(topicPartition.topic, topicPartition.partition).isEmpty)
```

Comment:
@lindong28 
* I agree that we could also filter out unknown partitions earlier, but the end result would be the same. The way it is now we seem to save some cycles if there is an error returned from `coordinator.handleFetchOffsets(...)`. Do you see other benefits in swapping the two parts?

* I wasn't too sure about the name `authorizedAndValidatedPartitionData` and welcome suggestions. It is meant to indicate all partitions that actually exist and are the user is authorized to see. Maybe `authorizedExistingPartitionData`? I don't know :)

* Thanks for the suggestion on moving the logic into GroupCoordinator. It makes sense, and if others also think that's the right thing to do for this PR (or in a separate one) I can work on that too.